feat: primeiro commit

sebastianwebber opened the pull request:
Este PR define a infra basica pra buildar uma imagem de VM no GCP.
Fica pendente a criacao da imagem usando ansible como provisionador.

Signed-off-by: Sebastian Webber <sebastian@swebber.me>
Co-authored-by: Afonso Rodrigues <afonsoaugustoventura@gmail.com>
Co-authored-by: Marcelo Mansur <marcelomansur@users.noreply.github.com>

sebastianwebber committed:
feat: primeiro commit

Este commit define a infra basica pra buildar uma imagem de VM no GCP.
Fica pendente a criacao da imagem usando ansible como provisionador.

Signed-off-by: Sebastian Webber <sebastian@swebber.me>
Co-authored-by: Afonso Rodrigues <afonsoaugustoventura@gmail.com>
Co-authored-by: Marcelo Mansur <marcelomansur@users.noreply.github.com>

gomex commented on Makefile:
```diff
@@ -20,9 +21,70 @@ ifdef INCLUDE_MAKEFILE
+		-e PACKER_PLUGIN_PATH=/workspace/.packer.d/plugins \
+		$(DOCKER_IMAGE) build packer/
+
+auth-create-sa: ## 1: create sa
+	gcloud iam service-accounts create ${GCP_PACKER_SA} \
+		--project ${GCP_PROJECT} \
+		--description="Packer Service Account" \
+		--display-name="Packer Service Account"
+
+auth-create-iam-policy: ## 2: create policy
+	gcloud projects add-iam-policy-binding ${GCP_PROJECT} \
+    --member=serviceAccount:${GCP_PACKER_SA}@${GCP_PROJECT}.iam.gserviceaccount.com \
+    --role=roles/compute.instanceAdmin.v1
+
+auth-create-add-policy: ## 3: add policy
+	gcloud projects add-iam-policy-binding ${GCP_PROJECT} \
+    --member=serviceAccount:${GCP_PACKER_SA}@${GCP_PROJECT}.iam.gserviceaccount.com \
+    --role=roles/iam.serviceAccountUser
+
+auth-create-creds-file: ## 4: generate creds file
+	gcloud iam service-accounts \
+		keys create creds.json \
+		--iam-account=${GCP_PACKER_SA}@${GCP_PROJECT}.iam.gserviceaccount.com
+
+generate-credentials: auth-create-sa auth-create-iam-policy auth-create-add-policy auth-create-creds-file ## generate file to creatential gcp
```
É possível colocar isso dentro de um container? Pra não precisar instalar nada na máquina.

sebawebber replied:
ajustado.

gomex commented on README.md:
```diff
@@ -1,18 +1,27 @@
 
 | Nome      	| Versao 	|
 |--------------	|---------	|
 |   Packer      |  >=1.7    |
```
Packer não é requisito. Docker é, não?

gomex commented on packer/source.pkr.hcl:
```diff
@@ -1,3 +1,9 @@
-source "" "" {
+source "googlecompute" "basic-example" {
+  project_id = "dadosabertosfeira" ## TODO: change-me
+  source_image = "ubuntu-2004-focal-v20210610"
+  ## ubuntu?
+  #ssh_username = "ubuntu"
+  ssh_username = "packer" ## vai funcionar?
 
+  zone = "us-central1-a"
```
Seria legal usar variável aqui, não?

gomex reviewed (changes requested):
Eu adorei o começo desse arquivo packer, mas acho que temos algumas coisas pra deixar ele melhor ainda.

sebastianwebber committed:
feat: adicionado editorconfig

Este commit adiciona a configuração do editor config, extraida do repo
https://github.com/mentoriaiac/iac-role-node-rke.

Também foi removido o arquivo `packer.pkr.hcl`, que era um exemplo e não
tinha uso.

Revisado por:
 - snifbr
 - viniciuscursi

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber committed:
fix: ajustes do packer no makefile

Este commit ajusta as chamadas do packer no docker, adicionando o env
file pra todas as chamadas.

Também movi a variavel `PACKER_PLUGIN_PATH` para o arquivo `.env`.

Revisado por:
  - snifbr
  - viniciuscursi

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber committed:
fix: adicionado plugin gcp

Este commit adiciona o plugin do `googlecompute` que estava faltando.
Sem ele o `packer build` da erros de autenticação.

Revisado por:
  - snifbr
  - viniciuscursi

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber committed:
chore: ajustado variaveis e nome

Este commit ajusta as configurações do source do `googlecompute`, tais
como seu nome e também adicionado variaveis com valores padrão, em
especial uma carregada do arquivo `.env`, a variavel de ambiente
`GCP_PROJECT` no caso.

Revisado por:
  - snifbr
  - viniciuscursi

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber committed:
feat: adicionado o playbook ansible

Este commit faz os ajustes necessários para executar a role
`iac-role-node-rke` na imagem do orquestrador criada pelo packer.

Atualizado a imagem do mansur para a ultima versão.

Também corrigido o problema do packer rodar no docker, que faltava a
configuração do `user` no arquivo `packer/build.pkr.hcl`.

Revisado por:
  - snifbr
  - vinicius-cursi
  - luizaoqui
  - lipenodias

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber commented:
IMPORTANTE: código ta quebrado. mandei detalhes no discord a mentoria. vamos continuar durante a semana.

sebastianwebber committed:
chore: adicionado role do nomad

Este commit adiciona a role do nomad na criação da imagem e também força
a utilização da versão v0.1.0.

Revisado por:
 - snifbr
 - vinicius-cursi
 - luizaoqui
 - lipenodias
 - gomex

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

sebastianwebber committed:
fix: checks do CI

Este commit ajusta os checks de sintaxe, apontando o mesmo para o
diretório `./packer/`.

Revisado por:
 - snifbr
 - vinicius-cursi
 - luizaoqui
 - lipenodias
 - gomex

Ref #2

Signed-off-by: Sebastian Webber <sebastian@swebber.me>

snifbr commented on packer/variables.pkr.hcl:
```diff
@@ -1,3 +1,14 @@
-variables {
-
+variable "config_gcp" {
+  type = object({
+    id_projeto     = string
+    zona           = string
+    familia_imagem = string
+    usuario_ssh    = string
+  })
+  default = {
+    id_projeto     = env("GCP_PROJECT")
+    zona           = "us-central1-a"
+    familia_imagem = "ubuntu-2004-lts"
+    usuario_ssh    = "mentoria_iac"
+  }
```
Individualizar esse bloco de variáveis, para ter melhor controle sobre o uso delas e não ter que declarar todas variáveis toda vez que desejar alterar somente um valor de uma variável.

gomex replied:
Vamos trabalhar nisso em outro PR ;) Valeu pelo report.
